json_serializable/lib/src/generator_helper.dart

```diff
@@ -85,28 +86,44 @@ class GeneratorHelper extends HelperCore with EncodeHelper, DecodeHelper {
       final createResult = createFactory(accessibleFields, unavailableReasons);
       yield createResult.output;
 
-      accessibleFieldSet = accessibleFields.entries
+      final fieldsToUse = accessibleFields.entries
           .where((e) => createResult.usedFields.contains(e.key))
           .map((e) => e.value)
-          .toSet();
+          .toList();
+
+      for (var candidate in sortedFields.where((element) =>
+          jsonKeyFor(element).explicitYesToJson &&
+          !fieldsToUse.contains(element))) {
+        fieldsToUse.add(candidate);
```

Comment:
Need to add candidates BACK even if they are not used in the factory if they are forced to be used for toJSON

Reply:
make these comments actual comments?

Reply:
done!